### PR TITLE
Feature: reusing workflows for testing / preview / release

### PR DESCRIPTION
**Pull request by TrueBrain**
Also removed all old actions and flows; they should no longer be used when switching over to v4.

This completely reworks how we deal with GitHub workflows. Basically, for most of the repositories, they all end up here with very little code, and we can maintain them all from this one place. There are risks involved with that of course, but hopefully the benefits outweighs that.

Example of how the workflows in repositories are going to look can be found here:

https://github.com/TrueBrain/sandbox/tree/main/.github/workflows

The three important workflows start with `rw-entry-`:

- `testing`: handles all the testing for a Docker / Python projects.
- `release`: handles the release when merging to main.
- `preview`: ensures only a single PR has a `preview` label, and the one that has is deployed to preview.

In the `sandbox` repository above I did most of the testing, and everything seems to work fine.

### CODE REVIEW

File: .github/workflows/rw-preview-label.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "https://api.github.com/repos/${{ github.repository }}/issues/${pr}/labels/preview"
+        done
+
+    # When two PRs get the Preview label before they can execute, the first will
+    # remove the label from the second, and the second from the first. This would
+    # leave no PR with a Preview label. So always make sure that if we finish,
+    # our PR has a preview label.
```
Comment (glx22) [inline .github/workflows/rw-preview-label.yml:39]: Doesn't that means both PR will get `preview` label in the end ?

Comment (TrueBrain) [reply .github/workflows/rw-preview-label.yml]: Added some extra words to make that a bit more clear :)

File: .github/workflows/rw-entry-release-docker-nomad.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+on:
+  workflow_call:
+    inputs:
+      service:
+        description: Name of the service on Nomad
+        required: true
+        type: string
+      source-root:
+        description: Root location of the project (where requirements.txt and Dockerfile are)
```
Comment (glx22) [inline .github/workflows/rw-entry-release-docker-nomad.yml:15]:
```suggestion
        description: Root location of the project (where Dockerfile is)
```

File: .github/workflows/rw-entry-preview-docker-nomad.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+on:
+  workflow_call:
+    inputs:
+      service:
+        description: Name of the service on Nomad
+        required: true
+        type: string
+      source-root:
+        description: Root location of the project (where requirements.txt and Dockerfile are)
```
Comment (glx22) [inline .github/workflows/rw-entry-preview-docker-nomad.yml:15]:
```suggestion
        description: Root location of the project (where Dockerfile is)
```

**Review by glx22 — APPROVED** [review]
Time to do tests in production :)

**Pull request merged**